### DISCUSSION TITLE
Custom operation extensions don't work

### DISCUSSION THREAD

**Discussion by nodeworks** [directus/directus#13720] [Archive]
### Describe the Bug

If I create a simple operation extension under the path: extensions/operations/test/index.js

```
"use strict";var e={id:"test",handler:async({milliseconds:e})=>{await new Promise((s=>setTimeout(s,Number(e))))}};module.exports=e;
```

I get error messages:

![Screen Shot 2022-06-03 at 12 47 53 PM](https://user-images.githubusercontent.com/5272855/171909659-2c6d9fb4-bb47-449d-9132-a6cbf250922c.png)

### To Reproduce

Create a simple extension under operations

### Errors Shown

![Screen Shot 2022-06-03 at 12 47 53 PM](https://user-images.githubusercontent.com/5272855/171909687-c769e170-e745-47de-9dc0-2ae717c14a1c.png)

### What version of Directus are you using?

9.12.0

### What version of Node.js are you using?

Directus 9.12.0 docker image node version

### What database are you using?

MS SQL

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
We haven't exposed the full custom operations extensions system yet (we're still wrapping up some loose ends + docs). That being said, this is the first operation type that requires two files: `api.js` and `app.js`. This is because there's both an API execution layer, and an app configuration layer.

**Reply by nodeworks** [reply]
@rijkvanzanten ah ok cool! but even with the two files it still errors out now but that's expected right? Until the custom operation extension compatibility is built out more?

**Reply by rijkvanzanten** [reply]
That's correct 🙂 The way the setup is designed is built with opening up the extension type in mind, but that's still a WIP (cc @nickrum)

**Reply by nodeworks** [reply]
Ok cool, I opened a PR here: https://github.com/directus/directus/pull/13741
